### PR TITLE
routing: deep copy any transaction we obtain from `GetBlock` call.

### PR DESCRIPTION
**Pull request by mohamedawnallah**
## Change Description
Deep copy any transaction we obtain from the `GetBlock` call.

Closes 2nd task in #5075

## Steps to Test
Steps for reviewers to follow to test the change.

### Testing
- [ ] Your PR passes all CI checks.
- [ ] Tests covering the positive and negative (error paths) are included.
- [ ] Bug fixes contain tests triggering the bug to prevent regressions.

### Code Style and Documentation
- [ ] The change obeys the [Code Documentation and Commenting](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#CodeDocumentation) guidelines, and lines wrap at 80.
- [ ] Commits follow the [Ideal Git Commit Structure](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#IdealGitCommitStructure).
- [ ] Any new logging statements use an appropriate subsystem and logging level.
- [ ] Any new lncli commands have appropriate tags in the comments for the rpc in the proto file.
- [ ]  [There is a change description in the release notes](https://github.com/lightningnetwork/lnd/tree/master/docs/release-notes), or `[skip ci]` in the commit message for small changes.

📝 Please see our  [Contribution Guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md) for further guidance.

### CODE REVIEW

File: lnwallet/test/test_interface.go [hunk 1]
```diff
@@ -1171,7 +1171,7 @@ func testListTransactionDetails(miner *rpctest.Harness,
 			transactions :=
 				make(map[chainhash.Hash][]*wire.TxOut, len(fetchedBlock.Transactions))
 			for _, tx := range fetchedBlock.Transactions {
-				transactions[tx.TxHash()] = tx.TxOut
+				transactions[tx.TxHash()] = tx.Copy().TxOut
```
Comment (guggero) [inline lnwallet/test/test_interface.go:1174]: nit: unrelated to the changes in this PR, but if we're touching this, can we format the line above to the following please?

```
			transactions := make(
				map[chainhash.Hash][]*wire.TxOut,
				len(fetchedBlock.Transactions),
			)
```

**Review by guggero — COMMENTED** [review]
Nice, thanks for the PR!

File: chainntnfs/interface.go [hunk 2]
```diff
@@ -773,7 +773,7 @@ func ConfDetailsFromTxIndex(chainConn TxIndexConn, r ConfRequest,
 	// Therefore we can save another query, and just use that height
 	// directly.
 	blockHeight, err := blockchain.ExtractCoinbaseHeight(
-		btcutil.NewTx(block.Transactions[0]),
+		btcutil.NewTx(block.Transactions[0].Copy()),
```
[the remark this answers is not in the document]
Comment (mohamedawnallah) [reply chainntnfs/interface.go:776]: Yes, this makes sense. ACK!

**Review by guggero — APPROVED** [review]
Thanks, LGTM :tada:

File: chainntnfs/bitcoindnotify/bitcoind.go [hunk 3]
```diff
@@ -875,10 +875,12 @@ func (b *BitcoindNotifier) historicalSpendDetails(
 			}
 
 			txHash := tx.TxHash()
+			txCopy := tx.Copy()
```
Comment (yyforyongyu) [inline chainntnfs/bitcoindnotify/bitcoind.go:878]: let's stick to the copy? So we should do `txHash := txCopy.TxHash()`

Comment (mohamedawnallah) [reply chainntnfs/bitcoindnotify/bitcoind.go:877]: Thanks! Updated

**Review by yyforyongyu — APPROVED** [review]

**Pull request merged**